src/index.ts

```diff
@@ -0,0 +1,129 @@
+import * as cm from "commonmark"
+import * as fm from "front-matter"
+import * as it from "@ts-common/iterator"
+
+export interface MarkDownEx {
+  readonly frontMatter?: string
+  readonly markDown: cm.Node
+}
+
+export type NodeType =
+  'text' | 'softbreak' | 'linebreak' | 'emph' | 'strong' | 'html_inline' | 'link' | 'image' | 'code' | 'document' | 'paragraph' |
+  'block_quote' | 'item' | 'list' | 'heading' | 'code_block' | 'html_block' | 'thematic_break' | 'custom_inline' | 'custom_block'
+
+export const createNode = (type: NodeType, ...children: cm.Node[]) => {
+  const result = new cm.Node(type)
+  for (const c of children) {
+    result.appendChild(c)
+  }
+  return result
+}
+
+export const createText = (literal: string) => {
+  const result = createNode("text")
+  result.literal = literal
+  return result
+}
+
+export const createCodeBlock = (info: string, literal: string) => {
+  const result = createNode("code_block")
+  result.info = info
+  result.literal = literal
+  return result
+}
```

Comment:
looks good, waiting for someone more familliar with markdown transformation for review